lists-service/src/main/java/au/org/ala/listsapi/service/ValidationService.java

```diff
@@ -0,0 +1,137 @@
+
+  @Value("${userDetails.api.url}")
+  private String userDetailsUrl;
+
+  private Map<String, List<ConstraintListItem>> constraints = null;
+
+  @PostConstruct
+  private void init() throws IOException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    File json = new File(constraintsFile);
+
+    constraints = objectMapper.readValue(json, new TypeReference<HashMap<String, List<ConstraintListItem>>>() {});
+
+    try {
+      List<Location> countries = fetchJson(userDetailsUrl + "ws/registration/countries.json", new TypeReference<>() {});
+
+      // Map the countries list into UI constraints
+      List<ConstraintListItem> countryConstraints = countries.stream().map(e -> {
+        var constraint = new ConstraintListItem();
+        constraint.setLabel(e.getName());
+        constraint.setValue(e.getIsoCode());
+
+        return constraint;
+      }).toList();
+
+      constraints.put("countries", countryConstraints);
+    } catch (Exception ex) {
+      log.error("Error loading country constraints from userdetails", ex);
+    }
+  }
+  private <T> T fetchJson(String uri, TypeReference<T> type) throws Exception {
+    HttpRequest httpRequest =
+            HttpRequest.newBuilder(new URI(uri))
+                    .GET()
+                    .build();
+
+    HttpResponse<String> response =
+            HttpClient.newBuilder()
+                    .proxy(ProxySelector.getDefault())
+                    .build()
+                    .send(httpRequest, HttpResponse.BodyHandlers.ofString());
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    return objectMapper.readValue(response.body(), type);
+  }
+
+  // Returns the constraint list as a map of List<ConstraintListItem> to correctly output JSON
+  public Map<String, List<ConstraintListItem>> getConstraintMap() {
+    return constraints;
+  }
+
+  public boolean isValueValid(ConstraintType constraintType, String value) {
+    List<ConstraintListItem> list = constraints.get(constraintType.name());
+
+    return list.stream()
+            .filter(elm -> elm.getValue().equals(value))
+            .findAny().orElse(null) != null;
+  }
+
+  public boolean isListValid(InputSpeciesList speciesList) {
+    // check that the supplied list type, region and license is valid
+    return (
+            isValueValid(ConstraintType.lists, speciesList.getListType()) &&
+            isValueValid(ConstraintType.licenses, speciesList.getLicence()) &&
+            isValueValid(ConstraintType.countries, speciesList.getRegion())
+    );
+  }
+
+  public List<ConstraintListItem> getConstraintList(ConstraintType constraintType) throws Exception {
+    List<ConstraintListItem> list = constraints.get(constraintType.name());
+
+    if (list == null) {
+      throw new Exception("Could not find corresponding constraint list for '" + constraintType + "' type!");
+    }
+
+    return list;
+  }
+
+  public List<ConstraintListItem> getConstraintRegions(String country) throws Exception {
```

Comment:
I cannot see where this is used by validation.

Reply:
Removed in 07271ef6f286a6a56b7fde4ac683bd9f59f5dc9a